src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/AvoidConstArraysTests.cs

```diff
@@ -806,28 +806,34 @@ private void M(string eventName, string msg)
         }
 
         [Fact, WorkItem(6686, "https://github.com/dotnet/roslyn-analyzers/issues/6686")]
-        public Task ArrayWithoutInitializer_Diagnostic()
+        public Task ArrayWithoutInitializer_NoDiagnostic()
         {
-            var source = @"using System.Collections.Generic;
+            return new VerifyCS.Test
+            {
+                TestCode = @"using System.Collections.Generic;
 
 public class MyClass
 {
-    public List<object> Cases => new() { {|CA1861:new object[0]|} };
-}";
-            var fixedSource = @"using System.Collections.Generic;
+    public List<object> Cases => new() { new object[0] };
```

Comment:
This should trigger CA1825 (https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1825).  Do we know why it doesn't?

Reply:
I think it is expected for unit testing, I did not see any unrelated diagnostics triggered in unit testing except compiler warnings. By my understanding the unit tests only trigger the diagnostics related to: https://github.com/dotnet/roslyn-analyzers/blob/340f69d38dec621a0d7d71e8f732afd6cc45cef2/src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/AvoidConstArraysTests.cs#L7-L8

In normal case the CA1825 is being triggered:
![image](https://github.com/dotnet/roslyn-analyzers/assets/45579687/a28e7ff5-7da5-4538-851f-75c0913d0077)